src/utils.rs

```diff
@@ -0,0 +1,40 @@
+use bincode::{self, Options};
+use serde::{Deserialize, Serialize};
+use std::fmt::Write;
+
+use crate::error::Result;
+
+pub fn serialize<T>(object: &T) -> Result<Vec<u8>>
+where
+    T: Serialize,
+{
+    let opts = bincode::DefaultOptions::new()
+        .with_little_endian()
+        .with_fixint_encoding();
+    Ok(opts.serialize(&object)?)
+}
+
+pub fn deserialize<'a, T>(buffer: &'a Vec<u8>) -> Result<T>
+where
+    T: Deserialize<'a>,
+{
+    let opts = bincode::DefaultOptions::new()
+        .with_little_endian()
+        .with_fixint_encoding();
+    let decoded: T = opts.deserialize(buffer.as_slice())?;
+
+    Ok(decoded)
+}
+
+pub fn arr_to_hex(bytes: &[u8]) -> String {
+    bytes.iter().fold(String::new(), |mut output, b| {
+        let _ = write!(output, "{b:02X}");
+        output
+    })
+}
```

Comment:
The hex crate you include in this project has an encode function which does exactly what the arr_to_hex functions intends to do.

Reply:
done